### PR TITLE
Discord Integration

### LINKED ISSUE #27: Parent system should be able to send messages to Discord

### PR DESCRIPTION
**Pull request by Michaelcraun**
## GitHub Ticket

https://github.com/mietechnologies/MIE-MCServer/projects/2#card-75093150
Closes #27 

## What changed?
- Added Discord messaging setup to initial setup
- Added functionality to message the Discord server when an unhandled exception is encountered if this setting is enabled

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Because there wasn't really a good place to put it, I added a new group to the config: 'Messaging'. This might be able to be used for more, in the future, for settings for posting to Slack or social media sites (if we get feature requests along those lines).

## Checklist
Please ensure the following items are taken care of as part of your PR. If they are not, your PR will NOT be approved!
- [x] You have added doc strings to any user-facing functions
- [x] You have updated the CHANGELOG.md file with your changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 - Added first draft of the README
+- During setup, the user now has the ability to configure settings to post messages to a Discord server
+- The project now messages a Discord server when it encounters an unhandled exception if this setting has been configured
```
Comment (BAChapin) [inline CHANGELOG.md:12]: It is generally good to only have 1 bullet point to summarize a single issue/card resolution. If we could condense this into a single point that is a bit more concise, I feel like that would be better overall.

Comment (Michaelcraun) [reply CHANGELOG.md:12]: I mean, I can, if you want me to, but they are kinda two separate ideas. 
1. Implemented specific functionality
2. Updated existing functionality to use new functionality
I guess I just don't understand why two separate pieces of work should be condensed into one bullet in the changelog when two different things were changed. If this is the case, I feel like we should be doing what the ticket says, no more, no less, which will lead to instances where we forget to update logic in other places that is affected by what we're doing in the moment. 

Lmk what you'd like me to do after hearing these justifications.

Comment (BAChapin) [reply CHANGELOG.md:12]: I feel like this might be an easier discussion to have offline or perhaps even on Slack.

Comment (Michaelcraun) [reply CHANGELOG.md:12]: I've responded on Slack several times to no avail, so I will continue the discussion here. Is the CHANGELOG supposed to be user- or dev-facing?

**Review by BAChapin — APPROVED** [review]

**Pull request merged**